plugin.video.cbc-sports/addon.py

```diff
@@ -0,0 +1,417 @@
+	    return  
+	rdata = str(get_html(url))
+	#try: mediaId = re.compile("mediaId': '(.+?)'").findall(str(data))[0]
+	if cbclog == 2:
+	    xbmc.log('CBC Sports rdata: ' + str(rdata), xbmc.LOGINFO)
+
+	try:					# primary mediaId parse
+	    startpos = rdata.find('mediaId')
+	    mediaId = rdata[startpos+8:startpos+21]
+	except IndexError:             
+	    xbmcgui.Dialog().notification(name, translation(30000), defaultimage, notetime, False)
+	    return
+	if cbclog >= 1:
+	    xbmc.log('CBC Sports mediaId: ' + mediaId, xbmc.LOGINFO)
+
+	try:					# primary mediaId verify int
+	    mediaIdint = int(mediaId)
+	except ValueError:			# alternate mediaId parse
+	    altid_start = rdata.find('identifier')
+	    mediaId = rdata[altid_start+13:altid_start+26]
+	    if cbclog >= 1:
+	        xbmc.log('CBC Sports alternate mediaId: ' + mediaId, xbmc.LOGINFO)
+
+	try:					# alternate mediaId verify int
+	    mediaIdint = int(mediaId)
+	except ValueError:
+	    xbmcgui.Dialog().notification(name, translation(30519), defaultimage, notetime, False)
+	    return
+	    
+	furl = basefeed + mediaId
+	jresponse = urllib.request.urlopen(furl)
+	jdata = json.load(jresponse)
+	if cbclog == 2:
+	    xbmc.log('CBC Sports Live Schedule Playback response: ' + str(jdata), xbmc.LOGINFO)
+
+	try:
+	    smil_url = jdata['entries'][0]['content'][0]['url']
+	except IndexError:             
+	    xbmcgui.Dialog().notification(name, translation(30000), defaultimage, notetime, False)
+	    if cbclog >= 1:
+	        xbmc.log('CBC Sports Live Schedule entry count is 0 for event: ' + str(name), xbmc.LOGINFO)
+	    return	
+
+	smil = get_html(smil_url)
+	contents = BeautifulSoup(smil,'html5lib')
+	if cbclog >= 1:
+	    xbmc.log('CBC Sports Live Schedule contents: ' + str(contents), xbmc.LOGINFO)
+					
+	if 'GeoLocationBlocked' in str(contents):		#  Check for blackout
+	    xbmcgui.Dialog().notification(name, translation(30001), defaultimage, notetime, False)
+	    return
+
+	try:
+		stream = (re.compile('video src="(.+?)"').findall(str(contents))[0]).replace('/z/','/i/').replace('manifest.f4m','master.m3u8')
+	except IndexError:			#  Check if stream is available
+		xbmcgui.Dialog().notification(name, translation(30000), defaultimage, notetime, False)
+		return
+
+	if cbclog >= 1:
+	    xbmc.log('CBC Sports Live Schedule Playback stream: ' + str(stream), xbmc.LOGINFO)
+	listitem = xbmcgui.ListItem(name)
+	listitem.setArt({'thumb': defaultimage, 'icon': defaultimage})
+	sdata = str(get_html(stream))
+
+	try:
+	    errfound = sdata.find('An error occurred')
+	except IndexError:
+            xbmcgui.Dialog().notification(name, translation(30010), defaultimage, notetime, False)
+            return
+
+	if errfound > -1:
+		xbmcgui.Dialog().notification(name, translation(30010), defaultimage, notetime, False)
+		return
+	if cbclog >= 1:
+	    xbmc.log('CBC Sports Live Schedule stream return data: ' + sdata, xbmc.LOGINFO)
+
+	try:	
+	    xbmc.Player().play( stream, listitem )
```

Comment:
Sorry, Player().play() is to be used by scripts not plugins. Please adjust this to use xbmcplugin.setResolvedUrl

Reply:
Code has been updated to use xbmcplugin.setResolvedUrl